apps/cms.c

```diff
@@ -62,168 +62,193 @@ struct cms_key_param_st {
 
 typedef enum OPTION_choice {
     OPT_COMMON,
-    OPT_INFORM, OPT_OUTFORM, OPT_IN, OPT_OUT, OPT_ENCRYPT,
-    OPT_DECRYPT, OPT_SIGN, OPT_CADES, OPT_SIGN_RECEIPT, OPT_RESIGN,
-    OPT_VERIFY, OPT_VERIFY_RETCODE, OPT_VERIFY_RECEIPT,
-    OPT_CMSOUT, OPT_DATA_OUT, OPT_DATA_CREATE, OPT_DIGEST_VERIFY,
-    OPT_DIGEST_CREATE, OPT_COMPRESS, OPT_UNCOMPRESS,
-    OPT_ED_DECRYPT, OPT_ED_ENCRYPT, OPT_DEBUG_DECRYPT, OPT_TEXT,
-    OPT_ASCIICRLF, OPT_NOINTERN, OPT_NOVERIFY, OPT_NOCERTS,
-    OPT_NOATTR, OPT_NODETACH, OPT_NOSMIMECAP, OPT_BINARY, OPT_KEYID,
-    OPT_NOSIGS, OPT_NO_CONTENT_VERIFY, OPT_NO_ATTR_VERIFY, OPT_INDEF,
-    OPT_NOINDEF, OPT_CRLFEOL, OPT_NOOUT, OPT_RR_PRINT,
-    OPT_RR_ALL, OPT_RR_FIRST, OPT_RCTFORM, OPT_CERTFILE, OPT_CAFILE,
-    OPT_CAPATH, OPT_CASTORE, OPT_NOCAPATH, OPT_NOCAFILE, OPT_NOCASTORE,
-    OPT_CONTENT, OPT_PRINT, OPT_NAMEOPT,
-    OPT_SECRETKEY, OPT_SECRETKEYID, OPT_PWRI_PASSWORD, OPT_ECONTENT_TYPE,
-    OPT_PASSIN, OPT_TO, OPT_FROM, OPT_SUBJECT, OPT_SIGNER, OPT_RECIP,
-    OPT_CERTSOUT, OPT_MD, OPT_INKEY, OPT_KEYFORM, OPT_KEYOPT, OPT_RR_FROM,
-    OPT_RR_TO, OPT_AES128_WRAP, OPT_AES192_WRAP, OPT_AES256_WRAP,
-    OPT_3DES_WRAP, OPT_WRAP, OPT_ENGINE,
+    OPT_IN, OPT_OUT,
+    OPT_CONFIG,
+
+    OPT_ENCRYPT, OPT_DECRYPT, OPT_SIGN, OPT_VERIFY, OPT_RESIGN,
+    OPT_SIGN_RECEIPT, OPT_VERIFY_RECEIPT,
+    OPT_DIGEST_CREATE, OPT_DIGEST_VERIFY, OPT_COMPRESS, OPT_UNCOMPRESS,
+    OPT_ED_ENCRYPT, OPT_ED_DECRYPT,
+    OPT_DATA_CREATE, OPT_DATA_OUT, OPT_CMSOUT,
+
+    OPT_INFORM, OPT_OUTFORM, OPT_RCTFORM,
+    /* OTP_STREAM = */ OPT_INDEF, OPT_NOINDEF,
+    OPT_BINARY, OPT_CRLFEOL, OPT_ASCIICRLF,
+
+    OPT_PWRI_PASSWORD, OPT_SECRETKEY, OPT_SECRETKEYID,
+    OPT_INKEY, OPT_PASSIN, OPT_KEYFORM, OPT_ENGINE, OPT_PROV_ENUM,
     OPT_R_ENUM,
-    OPT_PROV_ENUM, OPT_CONFIG,
-    OPT_V_ENUM,
-    OPT_CIPHER,
-    OPT_ORIGINATOR
+
+    OPT_ORIGINATOR, OPT_RECIP, /* OPT_PARAM, */ OPT_CIPHER,
+    OPT_WRAP, OPT_AES128_WRAP, OPT_AES192_WRAP, OPT_AES256_WRAP, OPT_3DES_WRAP,
+    OPT_DEBUG_DECRYPT,
+
+    OPT_MD, OPT_SIGNER, OPT_CERTFILE, OPT_CADES,
+    OPT_NODETACH, OPT_NOCERTS, OPT_NOATTR, OPT_NOSMIMECAP,
+    OPT_RR_ALL, OPT_RR_FIRST, OPT_RR_FROM, OPT_RR_TO,
+
+    /* OPT_SIGNER, */ OPT_CONTENT, OPT_NO_CONTENT_VERIFY, OPT_NO_ATTR_VERIFY,
+    OPT_NOSIGS, OPT_NOVERIFY, OPT_NOINTERN, /* OPT_CADES, */ OPT_VERIFY_RETCODE,
+    OPT_CAFILE, OPT_CAPATH, OPT_CASTORE,
+    OPT_NOCAFILE, OPT_NOCAPATH, OPT_NOCASTORE,
+
+    OPT_KEYOPT, OPT_KEYID, OPT_ECONTENT_TYPE,
+    OPT_TEXT, OPT_CERTSOUT,
+    OPT_TO, OPT_FROM, OPT_SUBJECT,
+
+    OPT_NOOUT, OPT_PRINT, OPT_NAMEOPT, OPT_RR_PRINT,
+
+    OPT_V_ENUM
```

Comment:
I suspected that the order of the option `enum` elements partially matters, but may be wrong on this.
And for maintaining the options I think it may help to keep them consistent, but it also adds a burden.
So reverted the changed order as you suggested.